src/main/kotlin/snyk/container/ContainerService.kt

```diff
@@ -2,18 +2,100 @@ package snyk.container
 
 import com.google.gson.Gson
 import com.google.gson.JsonSyntaxException
+import com.intellij.openapi.application.ReadAction
 import com.intellij.openapi.components.Service
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.fileEditor.FileDocumentManager
 import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.openapi.vfs.VirtualFileManager
 import io.snyk.plugin.cli.CliError
 import io.snyk.plugin.cli.ConsoleCommandRunner
+import io.snyk.plugin.getApplicationSettingsStateService
 import io.snyk.plugin.services.CliService
 import snyk.common.SnykError
+import snyk.iac.IacResult
+import java.nio.file.Paths
+
+private val LOG = logger<ContainerService>()
 
 @Service
 class ContainerService(project: Project) : CliService<ContainerResult>(
     project = project,
     cliCommands = listOf("container", "test")
 ) {
+
+    fun scan(iacResult: IacResult): ContainerResult {
+        LOG.info("starting scanning container service")
+
+        val issuesForFile = mutableListOf<ContainerIssuesForFile>()
+
+        iacResult.allCliIssues?.forEach { iacIssuesForFile ->
+            // at the moment we check only kubernetes manifest files
+            if (iacIssuesForFile.targetFile.endsWith("yaml") ||
+                iacIssuesForFile.targetFile.endsWith("yml")
+            ) {
+                val lineWithImage = ReadAction.compute<Pair<Int?, String>, RuntimeException> {
+                    findImageLineNumberOrNull(iacIssuesForFile.targetFile)
+                }
+
+                val containerIssueForFile = ContainerIssuesForFile()
+                containerIssueForFile.targetFile = iacIssuesForFile.targetFile
+                containerIssueForFile.lineNumber = lineWithImage.first!!
+
+                if (lineWithImage.first != null) {
+                    // extract image name
+                    val imageName = lineWithImage.second.trim().replace("image:", "").trim()
+                    LOG.info("Found image tag for container scan: $imageName")
+
+                    var commands = buildCliCommandsList()
+                    val mutCommands = commands.toMutableList()
+                    mutCommands.add(imageName)
+                    commands = mutCommands.toList()
```

Comment:
[nitpick]: Maybe we could simplify this like `commands = listOf(imageName, *commands.toTypedArray())`

Reply:
good catch :)